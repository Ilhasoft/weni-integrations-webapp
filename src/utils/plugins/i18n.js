import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
