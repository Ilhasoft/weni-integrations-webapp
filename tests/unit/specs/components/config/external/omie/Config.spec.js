import { unnnicCallAlert as mockUnnnicCallAlert } from '@weni/unnnic-system';

jest.mock('@weni/unnnic-system', () => ({
  ...jest.requireActual('@weni/unnnic-system'),
  unnnicCallAlert: jest.fn(),
}));

import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import OmieConfig from '@/components/config/external/omie/Config.vue';
import '@weni/unnnic-system';
import i18n from '@/utils/plugins/i18n';

import { singleApp } from '../../../../../../__mocks__/appMock.js';
singleApp.code = 'omie';

const localVue = createLocalVue();
localVue.use(Vuex);

const mountComponent = async ({
  loadingUpdateAppConfig = false,
  errorUpdateAppConfig = false,
} = {}) => {
  const state = {
    appType: {
      loadingUpdateAppConfig,
      errorUpdateAppConfig,
    },
  };

  const actions = {
    updateAppConfig: jest.fn(),
  };

  const store = new Vuex.Store({
    actions,
    state,
  });

  const wrapper = mount(OmieConfig, {
    localVue,
    store,
    i18n,
    propsData: {
      app: singleApp,
    },
  });

  await wrapper.vm.$nextTick();
  await jest.runAllTimers();

  return { wrapper, actions, state };
};

describe('components/config/external/omie/Config.vue', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', async () => {
    const { wrapper } = await mountComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateAppConfig with desired fields', async () => {
    const { wrapper, actions } = await mountComponent();

    const nameInput = wrapper.find('.app-config-omie__settings__content__inputs__name');
    const keyInput = wrapper.find('.app-config-omie__settings__content__inputs__key');
    const secretInput = wrapper.find('.app-config-omie__settings__content__inputs__secret');

    nameInput.vm.$emit('input', 'service name');
    keyInput.vm.$emit('input', 'service key');
    secretInput.vm.$emit('input', 'service secret');

    await wrapper.vm.$nextTick();

    expect(actions.updateAppConfig).not.toHaveBeenCalled();
    expect(mockUnnnicCallAlert).not.toHaveBeenCalled();

    const saveButton = wrapper.find('.app-config-omie__settings__buttons__save');
    saveButton.trigger('click');
    await wrapper.vm.$nextTick();

    expect(actions.updateAppConfig).toHaveBeenCalledWith(expect.any(Object), {
      code: 'omie',
      appUuid: singleApp.uuid,
      payload: {
        config: {
          name: 'service name',
          app_key: 'service key',
          app_secret: 'service secret',
        },
      },
    });
    expect(mockUnnnicCallAlert).toMatchSnapshot();
  });

  it('should call errorModal if update fails', async () => {
    const { wrapper, actions } = await mountComponent({ errorUpdateAppConfig: true });

    const nameInput = wrapper.find('.app-config-omie__settings__content__inputs__name');
    const keyInput = wrapper.find('.app-config-omie__settings__content__inputs__key');
    const secretInput = wrapper.find('.app-config-omie__settings__content__inputs__secret');

    nameInput.vm.$emit('input', 'service name');
    keyInput.vm.$emit('input', 'service key');
    secretInput.vm.$emit('input', 'service secret');

    await wrapper.vm.$nextTick();

    expect(actions.updateAppConfig).not.toHaveBeenCalled();
    expect(mockUnnnicCallAlert).not.toHaveBeenCalled();

    const saveButton = wrapper.find('.app-config-omie__settings__buttons__save');
    saveButton.trigger('click');
    await wrapper.vm.$nextTick();

    expect(actions.updateAppConfig).toHaveBeenCalledWith(expect.any(Object), {
      code: 'omie',
      appUuid: singleApp.uuid,
      payload: {
        config: {
          name: 'service name',
          app_key: 'service key',
          app_secret: 'service secret',
        },
      },
    });
    expect(mockUnnnicCallAlert).toMatchSnapshot();
  });

  describe('close configModal', () => {
    it('should emit a close event', async () => {
      const { wrapper } = await mountComponent({ errorUpdateAppConfig: true });

      const closeButton = wrapper.find('.app-config-omie__settings__buttons__cancel');
      closeButton.trigger('click');

      const event = wrapper.emitted('closeModal');

      expect(event).toBeTruthy();
      expect(event.length).toBe(1);
    });
  });
});
