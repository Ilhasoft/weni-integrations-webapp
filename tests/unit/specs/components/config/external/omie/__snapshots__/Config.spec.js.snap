// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/config/external/omie/Config.vue should be rendered properly 1`] = `
<div class="app-config-omie">
  <div class="app-config-omie__header">
    <div class="app-config-omie__header__title">
      <div class="app-config-omie__header__title__icon-container"><img src="https://weni-sp-push-dev.s3.sa-east-1.amazonaws.com/svg/logo-wwc.svg" class="app-config-omie__header__title__icon-container__icon"></div>
      <div class="app-config-omie__header__title__name">Weni Web Chat</div>
    </div> <span class="app-config-omie__header__description">Learn more about integrating with Omie <a href="https://ajuda.omie.com.br/pt-BR/articles/499061-obtendo-a-chave-de-acesso-para-integracoes-de-api" target="_blank" class="link">here</a></span>
  </div>
  <div class="app-config-omie__settings__content">
    <div data-v-1727ddfc="" class="config-omie__tabs tab size-md">
      <header data-v-1727ddfc="" class="tab-header">
        <ul data-v-1727ddfc="" class="tab-content">
          <li data-v-1727ddfc="" class="tab-head tab-head--active"> Configure </li>
        </ul>
      </header>
      <main data-v-1727ddfc="" class="tab-body">
        <div data-v-1727ddfc="" class="tab-panel">
          <div data-v-1727ddfc="">
            <div data-v-9ef23630="" class="app-config-omie__settings__content__inputs__name unnnic-form md" data-v-1727ddfc="">
              <p data-v-9ef23630="" class="unnnic-form__label">Name</p>
              <div data-v-8d0754d6="" data-v-9ef23630="" style="position: relative;" v-bind="[object Object]" mask="" class="unnnic-form-input"><input data-v-a95503a2="" data-v-8d0754d6="" type="text" size="0" v-bind="[object Object]" placeholder="" class="input size-md normal">
                <!---->
                <!---->
              </div>
              <!---->
            </div>
            <div data-v-9ef23630="" class="app-config-omie__settings__content__inputs__key unnnic-form md" data-v-1727ddfc="">
              <p data-v-9ef23630="" class="unnnic-form__label">App Key</p>
              <div data-v-8d0754d6="" data-v-9ef23630="" style="position: relative;" v-bind="[object Object]" mask="" class="unnnic-form-input"><input data-v-a95503a2="" data-v-8d0754d6="" type="text" size="0" v-bind="[object Object]" placeholder="" class="input size-md normal">
                <!---->
                <!---->
              </div>
              <!---->
            </div>
            <div data-v-9ef23630="" class="app-config-omie__settings__content__inputs__secret unnnic-form md" data-v-1727ddfc="">
              <p data-v-9ef23630="" class="unnnic-form__label">App Secret</p>
              <div data-v-8d0754d6="" data-v-9ef23630="" style="position: relative;" v-bind="[object Object]" mask="" class="unnnic-form-input"><input data-v-a95503a2="" data-v-8d0754d6="" type="text" size="0" v-bind="[object Object]" placeholder="" class="input size-md normal">
                <!---->
                <!---->
              </div>
              <!---->
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
  <div class="app-config-omie__settings__buttons"><button data-v-00f478da="" class="app-config-omie__settings__buttons__cancel unnnic-button unnnic-button--size-large unnnic-button--terciary unnnic-button-scheme--">
      <!---->
      <!---->
      <!----><span data-v-00f478da="" class="unnnic-button__label"> Cancel </span>
      <!---->
    </button> <button data-v-00f478da="" class="app-config-omie__settings__buttons__save unnnic-button unnnic-button--size-large unnnic-button--secondary unnnic-button-scheme--">
      <!---->
      <!---->
      <!----><span data-v-00f478da="" class="unnnic-button__label"> Connect </span>
      <!---->
    </button></div>
</div>
`;

exports[`components/config/external/omie/Config.vue should call errorModal if update fails 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "props": Object {
          "closeText": "Close",
          "icon": "alert-circle-1",
          "position": "bottom-right",
          "scheme": "feedback-red",
          "text": "An error occurred while trying to configure the service, please try again later",
          "title": "Error",
        },
        "seconds": 6,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`components/config/external/omie/Config.vue should call updateAppConfig with desired fields 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "props": Object {
          "closeText": "Close",
          "icon": "check-circle-1-1",
          "position": "bottom-right",
          "scheme": "feedback-green",
          "text": "Successfully configured service",
          "title": "Success",
        },
        "seconds": 6,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
